provider/resource_playbook.go

```diff
@@ -0,0 +1,574 @@
+		}
+
+		args = append(args, "--vault-id")
+
+		vaultIDArg := ""
+		if vaultID != "" {
+			vaultIDArg += vaultID
+		}
+
+		if vaultPasswordFile != "" {
+			vaultIDArg += "@" + vaultPasswordFile
+		} else {
+			log.Fatal("ERROR [ansible-playbook]: can't access vault file(s)! Missing 'vault_password_file'!")
+		}
+
+		args = append(args, vaultIDArg)
+	}
+
+	if len(extraVars) != 0 {
+		for key, val := range extraVars {
+			tmpVal, okay := val.(string)
+			if !okay {
+				log.Fatal("ERROR [ansible-playbook]: couldn't assert type: string")
+			}
+
+			args = append(args, "-e", key+"="+tmpVal)
+		}
+	}
+
+	args = append(args, playbook)
+
+	// set up the args
+	log.Print("[ANSIBLE ARGS]:")
+	log.Print(args)
+
+	if err := data.Set("args", args); err != nil {
+		log.Fatalf("ERROR [ansible-playbook]: couldn't set 'args'! %v", err)
+	}
+
+	if err := data.Set("env_vars", envVars); err != nil {
+		log.Fatalf("ERROR [ansible-playbook]: couldn't set 'env_vars'! %v", err)
+	}
+
+	return resourcePlaybookUpdate(data, meta)
+}
+
+func resourcePlaybookRead(data *schema.ResourceData, meta interface{}) error {
+	// Make sure an inventory exists
+	tempInventoryFile, okay := data.Get("temp_inventory_file").(string)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'temp_inventory_file'!", ansiblePlaybook)
+	}
+
+	_, tempInventoryFileInfoErr := os.Stat(tempInventoryFile)
+	if os.IsNotExist(tempInventoryFileInfoErr) {
+		return resourcePlaybookUpdate(data, meta)
+	}
+
+	/* ================================= */
+
+	ansiblePlaybookBinary, okay := data.Get("ansible_playbook_binary").(string)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'ansible_playbook_binary'!", ansiblePlaybook)
+	}
+
+	playbook, okay := data.Get("playbook").(string)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'playbook'!", ansiblePlaybook)
+	}
+
+	log.Printf("LOG [ansible-playbook]: playbook = %s", playbook)
+
+	argsTf, okay := data.Get("args").([]interface{})
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'args'!", ansiblePlaybook)
+	}
+
+	replayable, okay := data.Get("replayable").(bool)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'replayable'!", ansiblePlaybook)
+	}
+
+	ignorePlaybookFailure, okay := data.Get("ignore_playbook_failure").(bool)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'ignore_playbook_failure'!", ansiblePlaybook)
+	}
+
+	playFirstTime, okay := data.Get("play_first_time").(bool)
+	if !okay {
+		log.Fatalf("ERROR [%s]: couldn't get 'play_first_time'!", ansiblePlaybook)
+	}
+
+	if playFirstTime || replayable {
+		args := []string{}
+
+		// Get the rest of args
+		for _, arg := range argsTf {
+			tmpArg, okay := arg.(string)
+			if !okay {
+				log.Fatal("ERROR [ansible-playbook]: couldn't assert type: string")
+			}
+
+			args = append(args, tmpArg)
+		}
+
+		runAnsiblePlay := exec.Command(ansiblePlaybookBinary, args...)
+
+		runAnsiblePlayOut, runAnsiblePlayErr := runAnsiblePlay.CombinedOutput()
```

Comment:
The fact that this is running the playbook in the `Read` callback means the playbook is being run even during a `terraform plan` when `replayable = true`. I feel like we only want the playbook running during an apply.

Reply:
I've checked, and yes, that is correct. But, by putting the run into ``Create`` callback instead, the playbook will be run only on every first ``terraform apply`` (and maybe after every ``terraform destroy``too)... I will try moving this functionality into the Update function instead and see if this resolves it.

Reply:
I found a way to resolve this issue. Now the playbook shouldn't be running on ``terraform plan``.